src/common.ts

```diff
@@ -724,10 +724,11 @@ export function addVarName(
         currentScope.scopeError = recordScopeError(parser, Errors.DuplicateBinding, name);
       }
     }
-    if (value & BindingKind.CatchIdentifierOrPattern) {
-      if ((context & Context.OptionsWebCompat) === 0 || (value & BindingKind.CatchIdentifier) === 0) {
-        report(parser, Errors.DuplicateBinding, name);
-      }
+    if (
+      value & BindingKind.CatchPattern ||
+      (value & BindingKind.CatchIdentifier && (context & Context.OptionsWebCompat) === 0)
```

Comment:
@3cp I didn't use bitmask much, but why are we using it for options? Shouldn't we use boolean instead?

Reply:
Context uses bitmask for all options, as options can be applied together.